### PR TITLE
feat: add FastAPI template generation with all utilities

### PR DESCRIPTION
**Pull request by pythonhubdev**
## Overview

This PR significantly enhances the scaffoldr CLI tool by adding comprehensive Azure cloud support, improving VS Code development experience with dev containers, and ensuring robust testing across all configuration combinations. The tool now generates production-ready FastAPI applications with full cloud provider support (AWS S3, GCP Cloud Storage, Azure Blob Storage) and complete development environment setup.

## Type of Change
- [x] Feature

## Scope / Affected Areas
- [x] Developer tooling

## Breaking Changes
- [x] No

## Public API Impact (Python SDK)

- New modules/classes/functions:
  - `AzureStorageService` class in `services/azure_service.py`
  - `devcontainer.json` configuration for VS Code
- Changed/removed APIs:
- Deprecations (timeline and migration path):

## Testing & Verification

- Commands run locally:
	- [x] task test (pytest) - All tests pass across all configurations
	- [x] task lint (ruff/pyright/mypy) - Code formatting and linting verified
	- [x] pre-commit run --all-files (optional) - Pre-commit hooks validated
- Added/updated tests: [x] unit [x] integration
- Key scenarios and edge cases:
  - All configuration combinations tested (8 permutations)
  - Cloud type prompting validation
  - Edge cases: existing directories, invalid frameworks, custom destinations
  - Generated app testing with 100% pass rate

## Screenshots / Logs (optional)

```
✅ Project 'test-full' generated successfully!
📁 Change to the project directory: cd test-full
🚀 Run the application: uv run test_full

Test Results:
╭────────── Summary ──────────╮
│  13  Total Tests            │
│  13  Passed       (100.0%)  │
╰─────────────────────────────╯
```

## Documentation
- Notes: Updated internal documentation and code comments for Azure implementation

## Environment / Config Changes

- New/changed env vars:
  - `TEST_PROJECT_CLOUD_ACCOUNT_NAME` (Azure)
  - `TEST_PROJECT_CLOUD_ACCOUNT_KEY` (Azure)
  - `TEST_PROJECT_CLOUD_CONTAINER_NAME` (Azure)
- Defaults and validation: All cloud configurations include dummy values for development
- Backwards compatible: [yes]

## Checklist

- [x] Code formatted and linted (ruff format/check)
- [x] Type checks pass (pyright, mypy)
- [x] Tests pass locally (pytest)
- [x] CI is green (format/lint/type)
- [x] No secrets/keys committed
- [x] Linked issues updated and Acceptance Criteria met

### 🚀 New Features

1. **Azure Blob Storage Support**
   - Complete Azure Blob Storage service implementation
   - SAS URL generation for secure file access
   - Async operations with proper error handling
   - Environment configuration for Azure credentials

3. **Enhanced Cloud Type Prompting**
   - Automatic cloud provider selection when `use_cloud=true`
   - Support for AWS S3, GCP Cloud Storage, and Azure Blob Storage
   - Validation and error handling for invalid selections

### 🔧 Technical Improvements

1. **Template System Enhancements**
   - Fixed import issues in Azure service template
   - Added conditional dependency management
   - Improved configuration validation
   - Enhanced error handling in service implementations

2. **Testing Infrastructure**
   - Comprehensive test coverage for all configurations
   - Edge case validation (existing directories, invalid inputs)
   - Generated application testing verification
   - Cross-platform compatibility testing

3. **Developer Experience**
   - Improved CLI prompts and user feedback
   - Better error messages and validation
   - Consistent project structure across all configurations
   - Enhanced documentation and code comments

### 📦 Dependencies Added

- `azure-storage-blob>=12.23.1` - Azure Blob Storage client
- `azure-identity>=1.19.0` - Azure authentication

### 🧪 Test Coverage

- **8 Configuration Combinations Tested**:
  - Minimal (no extras)
  - Docker only
  - Cloud only (AWS/GCP/Azure)
  - Database only
  - Docker + Cloud
  - Docker + Database
  - Cloud + Database
  - Full stack (Docker + Cloud + Database)

- **100% Test Pass Rate** across all generated applications
- **Edge Cases Covered**: directory conflicts, invalid inputs, custom destinations

### 🎯 Impact

This PR transforms scaffoldr from a basic project generator into a production-ready, enterprise-grade scaffolding tool that supports:

- **Multi-cloud deployments** (AWS, GCP, Azure)
- **Containerized development** with VS Code integration
- **Database-backed applications** with proper ORM setup
- **Comprehensive testing** and validation
- **Developer-friendly tooling** with hot reload and debugging

The foundation is now solid for extending support to other frameworks (Litestar, Flask, etc.) and additional cloud providers in future PRs.

### CODE REVIEW

File: templates/fastapi_template/{{ project_name }}/src/{{ project_slug }}/services/__init__.py.jinja [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from {{ project_slug }}.services.repositories.storage_repository import (
+	FileInfo,
+	StorageRepository,
```
[the remark this answers is not in the document]
Comment (pythonhubdev) [reply templates/fastapi_template/{{ project_name }}/src/{{ project_slug }}/services/__init__.py.jinja:3]: Fixed
